ibis/backends/tests/test_sql.py

```diff
@@ -195,3 +195,14 @@ def test_union_aliasing(backend_name, snapshot):
     result = top_ten.union(bottom_ten)
 
     snapshot.assert_match(str(ibis.to_sql(result, dialect=backend_name)), "out.sql")
+
+
+def test_union_generates_predictable_aliases(con):
```

Comment:
So this case was failing to run on `main`, right?

Reply:
Yep:

```
In [3]: import ibis
   ...:
   ...: from ibis import _
   ...:
   ...: t = ibis.examples.penguins.fetch()
   ...: # No error if you remove the .head() call
   ...: t = t.head(100)
   ...:
   ...: # The rename is simply needed so the schemas are the same so the union works
   ...: sub1 = t.inner_join(t.view(), "island").mutate(island_right=_.island)
   ...: sub2 = t.inner_join(t.view(), "sex").mutate(sex_right=_.sex)
   ...: u = ibis.union(sub1, sub2)
   ...: u.execute()
...
CompileError: Multiple, unrelated CTEs found with the same name: 't5'
```